### PR TITLE
Also use Javascript as language in l10n.pl - fixes #13924

### LINKED ISSUE #13924: L10N script fails to extract strings from JS files when using string concatenation
See https://github.com/owncloud/news/issues/719

The following code fails to be translated although its perfectly fine js
```js
t('news', 'a' +
              'string');
```

This makes it impossible to wrap longer strings. I think its due to the reason that we are using Python mode when parsing js files. In Python the above code would throw a SyntaxError so I guess that's the reason why it's parsed incorrectly.

Any ideas? We really need to go for something that can parse JS properly and the perl file is at least for me unreadable. Can we port that to something that can actually be edited by more than 2 people?

@PVince81 @jakobsack @DeepDiver1975 @MorrisJobke @LEDfan @georgehrke @nickvergessen @butonic

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #13924

### CODE REVIEW

File: l10n/l10n.pl [hunk 1]
```diff
@@ -74,6 +74,23 @@ sub getPluralInfo {
 	return $info;
 }
 
+sub init() {
+	# let's get the version from stdout of xgettext
+	my $out = `xgettext --version`;
+	# we assume the first line looks like this 'xgettext (GNU gettext-tools) 0.19.3'
+	$out = substr $out, 29, index($out, "\n")-29;
+	$out =~ s/^\s+|\s+$//g;
+	$out = "v" . $out;
```
Comment (jakobsack) [inline l10n/l10n.pl]: I'd use another code for getting the version number:
```perl
	$out = ( split /\n/, $out )[0];
	$out =~ /([\d\.]+)\s*$/;
	$out = "v" . $1;
```

This code the first line, then searches for numbers and dots at the end of the line.

An alternative to line 2 would be 
```perl
$out =~ /([^\s]+)\s*$/;
```
This code could handle "0.1.2-alpha" as well. I don't know if this is needed for xgettext.

Comment (MorrisJobke) [inline l10n/l10n.pl]: As soon as the dirty old perl code is touched our lovely @jakobsack is active again :smile:

**Comment by DeepDiver1975** [comment]
Ready to be reviewed @LukasReschke @MorrisJobke @blizzz @icewind1991 @schiesbn

**Comment by LukasReschke** [comment]
874c939233c387ca1a1aa84b727178c7585f1a8afbf385ed8484428695fa95ba

**Comment by LukasReschke** [comment]
Whatever. I have no idea about Perl but let's do :see_no_evil: :hear_no_evil: :speak_no_evil: 

:+1:

**Comment by blizzz** [comment]
same as @LukasReschke, otherwise looks OK as far as I can tell :+1:

**Comment by MorrisJobke** [comment]
Conflicts 💥

**Pull request merged**